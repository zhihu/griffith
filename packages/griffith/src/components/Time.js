import React from 'react'
import PropTypes from 'prop-types'
import formatTime from '../utils/formatTime'

class Time extends React.Component {
  static propTypes = {
    value: PropTypes.number,
  }

  static defaultProps = {
    value: 0,
  }

  render() {
    const {value} = this.props
    return <span>{formatTime(value)}</span>
  }
}

export default Time
