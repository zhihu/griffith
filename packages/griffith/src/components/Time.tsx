import React from 'react'
import formatTime from '../utils/formatTime'

type OwnProps = {
  value?: number
}

type Props = OwnProps & typeof Time.defaultProps

class Time extends React.Component<Props> {
  static defaultProps = {
    value: 0,
  }

  render() {
    const {value} = this.props
    return <span>{formatTime(value)}</span>
  }
}

export default Time
