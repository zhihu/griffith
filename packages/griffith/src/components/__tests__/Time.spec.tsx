import React from 'react'
import Renderer from 'react-test-renderer'
import Time from '../Time'

describe('Time', () => {
  it('get Time component', () => {
    expect(Renderer.create(<Time value={904} />).toJSON()).toMatchSnapshot()
    expect(Renderer.create(<Time />).toJSON()).toMatchSnapshot()
  })
})
