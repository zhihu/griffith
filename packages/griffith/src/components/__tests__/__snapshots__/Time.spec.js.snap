// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time get Time component 1`] = `
<span>
  15:04
</span>
`;

exports[`Time get Time component 2`] = `
<span>
  00:00
</span>
`;
